### PR TITLE
[Backport 2.x] Bump protobuf-java to 3.21.8

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
Backport #4886 to 2.x

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Update Apache Lucene to 9.4.1 ([#4922](https://github.com/opensearch-project/OpenSearch/pull/4922))
 - Bump `woodstox-core` to 6.4.0 ([#4951](https://github.com/opensearch-project/OpenSearch/pull/4951))
 - Upgrade jetty-http, kotlin-stdlib and snakeyaml ([#4982](https://github.com/opensearch-project/OpenSearch/pull/4982))
+- Bump protobuf-java to 3.21.8 ([#5005](https://github.com/opensearch-project/OpenSearch/pull/5005))
```
Comment (dblock) [inline CHANGELOG.md:43]: If I understand correctly this should be in the 2.x section? @kotwanikunal?

Comment (kotwanikunal) [reply CHANGELOG.md:43]: Yes!

Comment (VachaShah) [reply CHANGELOG.md]: Moved, thank you!

**Comment by VachaShah** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/6099/
> * **CommitID:** [55a4abd](https://github.com/opensearch-project/OpenSearch/commit/55a4abd1be8ae195e3498db04d0199ecd96a134c)
>   Please examine the workflow log, locate, and copy-paste the failure below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Failing tests for version assertions, an example:

```
org.opensearch.test.VersionUtilsTests.testResolveReleasedVersionsForUnreleasedStableBranch

java.lang.AssertionError: 
Expected: <[2.0.0, 3.0.0, 3.0.1]>
     but: was <[2.0.0, 3.0.0, 3.0.1, 3.0.2, 3.1.0]>
	at __randomizedtesting.SeedInfo.seed([360B543EE03021EF:287365EFC398C638]:0)

**Review by andrross — APPROVED** [review]

**Pull request merged**